newlib/libc/tinystdio/stdio_private.h

```diff
@@ -216,13 +216,56 @@ __stdio_sflags (const char *mode)
     return __posix_sflags (mode, &omode);
 }
 
-#else
+#else // POSIX_IO
 
 int
 __stdio_sflags (const char *mode);
 
+#endif // POSIX_IO
+
+#ifdef _WANT_FLOCKFILE
+#define FILE_FN_UNLOCKED_SPECIFIER static inline
+#define FILE_FN_UNLOCKED(_fn) _fn##_unlocked
+
+wint_t ungetwc_unlocked(wint_t c, FILE *stream);
+int ungetc_unlocked(int c, FILE *stream);
+int fgetc_unlocked(FILE *stream);
+
+#else
+#define FILE_FN_UNLOCKED_SPECIFIER
+#define FILE_FN_UNLOCKED(_fn) _fn
 #endif
 
+static inline void __flockfile(FILE *f) {
+	(void) f;
+#ifdef _WANT_FLOCKFILE
+	if (f->lock)
```

Comment:
Why does this test f->lock?

Reply:
it was a workaround for strings functions like `sprintf`

removed now